### ISSUE TITLE
vQFX factory default login has changed

### ISSUE THREAD

**Issue by aninchat** [hellt/vrnetlab#72]
on newer vQFX code, factory default login has changed. It is no longer root/Juniper. Instead, it is root and no password (simply a carriage return). Because of this, the container initial setup eventually fails. The bringup stalls at:

```
2022-02-03 14:43:57,110: launch     INFO     matched login prompt
2022-02-03 14:43:57,110: launch     DEBUG    writing to serial console: root
2022-02-03 14:43:57,110: launch     TRACE    Waiting for Password:
```

In earlier code (like 19.4), it would proceed correctly with root/Juniper:

```
2022-02-03 15:41:56,742: launch     INFO     matched login prompt
2022-02-03 15:41:56,742: launch     DEBUG    writing to serial console: root
2022-02-03 15:41:56,743: launch     TRACE    Waiting for Password:
2022-02-03 15:42:21,835: launch     TRACE    Read:  root
Password:
2022-02-03 15:42:21,835: launch     DEBUG    writing to serial console: Juniper
2022-02-03 15:42:24,839: launch     TRACE    OUTPUT VCP:

2022-02-03 15:42:33,067: launch     DEBUG    writing to serial console: cli
2022-02-03 15:42:33,068: launch     TRACE    Waiting for >
2022-02-03 15:42:38,651: launch     TRACE    Read:  cli
{master:0}
root@vqfx-re>
2022-02-03 15:42:38,651: launch     DEBUG    writing to serial console: configure
2022-02-03 15:42:38,651: launch     TRACE    Waiting for #
2022-02-03 15:42:38,723: launch     TRACE    Read:  configure
```

Can this new factory default login be accommodated please, as this essentially breaks vQFX in containerlab (for any new release)?

**Comment by hellt** [comment]
@chriscummings-esnet do you want to have a stab at it? I remember you had newest vqfx at your disposal

**Comment by aninchat** [comment]
From my testing of some of the new images, I think the following changes will be needed:

The snippet below will need to be adjusted:

```
        (ridx, match, res) = self.tn.expect([b"login:", b"root@vqfx-re:RE:0%"], 1)
        if match: # got a match!
            if ridx == 0: # matched login prompt, so should login
                self.logger.info("matched login prompt")
                self.wait_write("root", wait=None)
                self.wait_write("Juniper", wait="Password:")
```

The shell prompt is no longer 'root@vqfx-re:RE:0%'. It is 'root@:RE:0%'.

The next thing, in the same snippet - waiting for a password needs to be removed. In the newer code, as soon as you login as root during factory default settings, it puts you into shell. No password is needed (no carriage return is expected either).

**Comment by hellt** [comment]
@aninchat do you know with which version this new login routine/prompt changed?

**Comment by aninchat** [comment]
> @aninchat do you know with which version this new login routine/prompt changed?

I believe this is from 20.x, but I am not entirely sure. Currently, you can download 19.4 and 20.2 for free from Juniper. 19.4 requires root/Juniper for login, and 20.2 requires only root. So, from my perspective, 20.2 onwards this change should be made for containerlab.

**Comment by hellt** [comment]
@aninchat was the issue fixed when you donwload 20.2 from Juniper site and 19.4 sits within the archive?

**Comment by hellt** [comment]
I have merged fixes for vqfx to detect the version from the file and use appropriate login routine
Tested this with the real 20.2 image. Would be good for someone to test 19.4
